matchzoo/preprocessor/process_units.py

```diff
@@ -346,6 +346,50 @@ def transform(
         return hashing
 
 
+class SlidingWindowUnit(ProcessorUnit):
+    """
+    SlidingWindowUnit class.
+
+    Used to get information from a window sliding
+    through the input data.
+
+    Examples:
+        >>> data = np.array([[0,0,0],[1,1,1],[2,2,2],[3,3,3]])
+        >>> sliding = SlidingWindowUnit()
+        >>> output = sliding.transform(data)
+        >>> output.shape
+        (2, 9)
+        >>> output[0]
+        array([0, 0, 0, 1, 1, 1, 2, 2, 2])
+
+    """
+
+    def __init__(self, sliding_window: int=3):
+        """
+        Class initialization.
+
+        :param sliding_window: sliding window length.
+        """
+        self._sliding_window = sliding_window
+
+    def transform(self, inputs: np.ndarray) -> np.ndarray:
+        """
+        Compute the result of a window sliding through input data.
+
+        :param inputs: sequential input data.
+        :return: window sliding result.
+        """
+        output = list()
+        if len(inputs) < self._sliding_window:
+            raise ValueError(
+                'Input data can not fill user-defined window.')
```

Comment:
I guess the best thing to do is to use logging or warning listed of raise an Exception and use `pass` to continue